### ISSUE TITLE
Can't open control settings after editing control prefix

### ISSUE THREAD

**Issue by ebruchez** [orbeon/orbeon-forms#4288]
- insert Number
- Control Settings
- update prefix
- apply settings
- try Control Settings again
- dialog doesn't open

What happens after adding the prefix is that the control changes its id. For example in my case it coes from `control-3-control` to `xf-311`. When you try to open the the Control Settings again, Form Builder passes `xf-311` and the server can't find that control in the form definition.

**Comment by ebruchez** [comment]
This is because we remove the `id` attribute. Regressed with #1433.

**Comment by ebruchez** [comment]
- [ ] don't remove the `id` attribute
- [ ] more to fix as Control Settings doesn't read things like `prefix` anymore either

**Comment by ebruchez** [comment]
The way things work since that commit is that we delete attributes which don't exist on the view template in the XBL component. `fr:number` doesn't have for example `prefix` in that template.

So the question is whether we are happy with that and need to add all the attribute on the template. The drawback is that we get extra attributes in the form. Some might be not desirable, like `grouping-separator`, which I think have a different meaning when empty.

I am just wondering why we are deleting these attributes. To discuss with @avernet.

**Comment by ebruchez** [comment]
Following discussion with @avernet:

This was done so that when you change a control's appearance, you don't keep extra attributes which might be relevant to the older appearance. So a list of relevant attributes for an appearance is required.

Now, specifically, `fr:number` only has a single appearance. Could we make it so that in such cases we don't prune attributes? Or do this only when we prune appearances?

**Comment by ebruchez** [comment]
What #1433 did when inserting the control to the dynamic editor  is twofold:

- *add* attributes that are in the view template but not in the control (with the same value as on the template)
- *remove* attributes that are in the control but not in the template

It does this in `fb-process-xbl-fb-control-details`:

- when the dialog loads
- when the dialog moves from control to control (prev/next)
- when the value of the appearance changes in the dialog

**Comment by ebruchez** [comment]
Implicit notions:

- supported attributes
- required attributes

Right now, after #1433, the code assumes that all supported and required attributes are on the view template. `fr:number` assumes it does all the work on its own.

**Comment by ebruchez** [comment]
So what should happen if you have `fr:my-foo` and `fr:my-bar` which represent two appearances `foo` and `bar` of `fr:my`. But for some reason, `fr:my-foo` and `fr:my-bar` support different attribute sets.

I understand that we should probably *prune* unsupported attributes.

However, I am not sure we should *add* all missing attributes. It would be better for the editor to figure that out. Unless we have another notion of the attributes being required.

In short this is a bit of a headache.

**Comment by ebruchez** [comment]
As an immediate fix, what if we only add/remove the attributes when there is an appearance change?
